scoreboard/management/commands/wipe_db.py

```diff
@@ -1,30 +1,92 @@
 # Wipe everything in the database without changing the schema.
-from django.core.management.base import BaseCommand, CommandError
+from django.core.management.base import BaseCommand
 from django.db import transaction
 from scoreboard.models import *
 
+
+@transaction.atomic
+def wipe_leaderboard_fields(entity):
+    entity.longest_streak = 0
+    entity.unique_deaths = 0
+    entity.unique_ascs = 0
+    entity.unique_achievements = 0
+    entity.games_over_1000_turns = 0
+    entity.games_scummed = 0
+    entity.total_games = 0
+    entity.wins = 0
+    entity.trophies.clear()
+    entity.save()
+
+
+@transaction.atomic
+def clear_player_and_clan_fields():
+    for player in Player.objects.all():
+        wipe_leaderboard_fields(player)
+    for clan in Clan.objects.all():
+        wipe_leaderboard_fields(clan)
+
+
+@transaction.atomic
+def wipe_games():
+    Game.objects.all().delete()
+    clear_player_and_clan_fields()
+
+
+@transaction.atomic
+def wipe_all_but_clans():
+    Clan.objects.all().delete()
+    Player.objects.all().delete()
+    User.objects.all().delete()
+    clear_player_and_clan_fields()
```

Comment:
If we're wiping everything but clans, why is it calling `Clan.objects.all.delete()` (and Player, while not deleting Games)?

🤔

(Do we have a use case for where someone would want to wipe "everything but clans" anyway, deleting Achievement and Conduct and the fixtures and everything else? Maybe, like, pre-tournament where K2 wants to import an updated achievements.yaml file but clan signup is already open? In that case I'd expect this to delete all the OTHER models besides these three, and leave these three untouched.)

Reply:
good catch